### DISCUSSION TITLE
Can you link timeout SQEs to multishot recv?

### DISCUSSION THREAD

**Discussion by cmazakas** [axboe/liburing#904] [Q&A] [answered]
Does multishot recv() support `io_uring_prep_link_timeout`?

This leads to an even more general question, how do multishot ops work with chaining SQEs? Is the chain executed only once or does it happen every time a CQE comes in?

**Comment by isilence** [comment, marked as answer]
> Does multishot recv() support `io_uring_prep_link_timeout`?

I don't remember actually trying it out but it should work, i.e. trying to cancel the request when time comes. We'd need to check how racy it is though, the case I'm afraid of is if it can ignores cancellation and continue running.

> This leads to an even more general question, how do multishot ops work with chaining SQEs? Is the chain executed only once or does it happen every time a CQE comes in?

Linked timeouts work differently from normal links. The latter ones will be run only after the multishot is cancelled,  fails, etc..
